File: Backend (Springboot)/Annapurna/src/main/java/com/annapurna/domain/Order.java
package com.annapurna.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "orders")
public class Order {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private int quantity;
	private double totalPrice;	
	private LocalDateTime orderTime;

	@ManyToOne
	private User customer;

	@ManyToOne
	private Tiffin tiffin;

	@Enumerated(EnumType.STRING)
	@Column(length = 20)
	private OrderStatus status = OrderStatus.PENDING;

	@Enumerated(EnumType.STRING)
	@Column(length = 20)
	private PaymentStatus paymentStatus = PaymentStatus.PENDING;

	private String razorpayOrderId;
	private String razorpayReceipt;
	private String razorpayStatus;
	private String razorpayPaymentId;
}

File: Backend (DNET)/Annapurna/Models/Order.cs
using System.ComponentModel.DataAnnotations;

namespace AnnapurnaAPI.Models
{
    public class Order
    {
        [Key]
        public long Id { get; set; }

        public int Quantity { get; set; }
        public double TotalPrice { get; set; }
        public DateTime OrderTime { get; set; } = DateTime.UtcNow;

        // Navigation properties
        public virtual User Customer { get; set; } = null!;
        public long CustomerId { get; set; }  //  Changed to long

        public virtual Tiffin Tiffin { get; set; } = null!;
        public long TiffinId { get; set; }

        public OrderStatus Status { get; set; } = OrderStatus.PENDING;
        public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.PENDING;

        // Razorpay fields
        public string? RazorpayOrderId { get; set; }
        public string? RazorpayReceipt { get; set; }
        public string? RazorpayStatus { get; set; }
        public string? RazorpayPaymentId { get; set; }
    }
}
